### PR TITLE
Remove traits_super

### LINKED ISSUE #1013: Remove traits_super
This class was introduced a long time ago (older than Traits 2 or maybe even older):
https://github.com/enthought/traits/blob/2685c7a260228fe535582606c72f892dac8ccb9a/traits/has_traits.py#L3396-L3404

There are no evidence that suggests this is needed. We should remove it.

### PR DESCRIPTION
**Pull request by kitchoi**
Closes #1013

This PR removes `traits_super`.

**Checklist**
- ~Tests~: Never tested.
- [x] Update API reference (`docs/source/traits_api_reference`)
- ~Update User manual (`docs/source/traits_user_manual`)~ : Never mentioned
- [x] Update type annotation hints in `traits-stubs`

### CODE REVIEW

**Comment by kitchoi** [comment]
Unrelated note: Thanks to the checklist which reminded me of the docs and annotations. I had initially forgotten!

File: docs/source/traits_api_reference/has_traits.rst [hunk 1]
```diff
@@ -41,7 +41,6 @@ Classes
 
 .. autoclass:: ISerializable
 
-.. autoclass:: traits_super
```
Comment (mdickinson) [inline docs/source/traits_api_reference/has_traits.rst:44]: Do we want to lose another blank line here?

(I had to find _something_ to complain about. :-)

Comment (kitchoi) [reply docs/source/traits_api_reference/has_traits.rst:44]: Yes! (Who knows if I deliberately left it there for it to be complained about ;)  )

**Comment by mdickinson** [comment]
Still LGTM!

**Pull request merged**